packages/core/src/pubsub/__tests__/outstanding-queries.test.ts

```diff
@@ -0,0 +1,161 @@
+  beforeEach(() => {
+    outstandingQueries.queryMap.clear()
+    outstandingQueries.queryQueue.clear()
+  })
+
+  test('Oustanding Queries, Add Query, verify add cleans upon add', async () => {
+    const tOld = new Date('2022-05-12T10:20:30Z').getTime()
+    const qOld = new Query(tOld, FAKE_STREAM_ID, 'a')
+    const testQueryIDOld = 'testIDOld'
+    outstandingQueries.add(testQueryIDOld, qOld)
+    expect(outstandingQueries.queryMap.size).toEqual(1)
+    const tNew = Date.now()
+    const qNew = new Query(tNew, FAKE_STREAM_ID2, 'b')
+    const testQueryIDNew = 'testIDNew'
+    outstandingQueries.add(testQueryIDNew, qNew)
+    expect(outstandingQueries.queryQueue.size()).toEqual(1)
+  })
+
+  test('Oustanding Queries, Add Query now', async () => {
+    const t1 = Date.now()
+    const q1 = new Query(t1, FAKE_STREAM_ID, 'a')
+    const testQueryID = 'testID'
+    outstandingQueries.add(testQueryID, q1)
+    expect(outstandingQueries.queryMap.size).toEqual(1)
+    expect(outstandingQueries.queryQueue.size()).toEqual(1)
+  })
+
+  test('Oustanding Queries, Ensure Non-expired timestamp is not deleted', async () => {
+    const t1 = Date.now()
+    const q1 = new Query(t1, FAKE_STREAM_ID, 'a')
+    const testQueryID = 'testID'
+    outstandingQueries.add(testQueryID, q1)
+    const t2 = Date.now()
+    const q2 = new Query(t2, FAKE_STREAM_ID2, 'b')
+    const testQueryID2 = 'testIDNew'
+    outstandingQueries.add(testQueryID2, q2)
+    expect(outstandingQueries.queryMap.size).toEqual(2)
+    expect(outstandingQueries.queryQueue.size()).toEqual(2)
+  })
+
+  test('Oustanding Queries, Remove Query', async () => {
+    const t1 = new Date('2022-05-12T10:20:30Z').getTime()
+    const testQueryID = 'testID'
+    const q1 = new Query(t1, FAKE_STREAM_ID, testQueryID)
+    outstandingQueries.add(testQueryID, q1)
+    assert(outstandingQueries.queryMap.size === 1)
+    assert(outstandingQueries.queryQueue.size() === 1)
+    outstandingQueries.remove(q1)
+    expect(outstandingQueries.queryMap.size).toEqual(0)
+
+    expect(outstandingQueries.queryQueue.size()).toEqual(0)
+  })
+
+  test('Oustanding Queries, cleanUpExpiredQueries', async () => {
+    const t1 = new Date('2022-05-12T10:20:30Z').getTime()
+    const testQueryID = 'testID'
+    const q1 = new Query(t1, FAKE_STREAM_ID, testQueryID)
+    outstandingQueries.add(testQueryID, q1)
+    assert(outstandingQueries.queryMap.size === 1)
+    assert(outstandingQueries.queryQueue.size() === 1)
+    outstandingQueries.cleanUpExpiredQueries()
+    expect(outstandingQueries.queryMap.size).toEqual(0)
+    expect(outstandingQueries.queryQueue.size()).toEqual(0)
+  })
+
+  test('Oustanding Queries, restrict duplicate outstanding queries', async () => {
+    const t1 = new Date('2022-05-12T10:20:30Z').getTime()
+    const testQueryID = 'testID'
+    const q1 = new Query(t1, FAKE_STREAM_ID, testQueryID)
+    outstandingQueries.add(testQueryID, q1)
+    expect(!outstandingQueries.add(testQueryID, q1))
```

Comment:
the `add` method is a void method, so this `expect` check isn't really checking anything since `add` has no return value

Reply:
I think this test should be updated to test that when a duplicate query id is added, the timestamp in the set is updated to that of the newer query